src/indra_cogex/analysis/source_targets_explanation.py

```diff
@@ -551,27 +617,97 @@ def run_explain_downstream_analysis(source_hgnc_id, target_hgnc_ids, *, client):
     # Initialize results dictionary
     results = {}
 
+    # Create output directory if specified and it doesn't exist
+    if output_dir and not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
     # 1. Get statements and create visualizations
     stmts_df, filtered_df = get_stmts_from_source(source_hgnc_id, target_proteins=target_hgnc_ids)
 
     # Create and convert interaction plot
     interaction_fig = plot_stmts_by_type(filtered_df)
     results['interaction_plot'] = convert_plot_to_base64(interaction_fig)
 
-    results['statements'] = assemble_protein_stmt_htmls(filtered_df)
+    # Save interaction plot if output directory is specified
+    if output_dir:
+        # Save as PNG image
+        interaction_fig.savefig(os.path.join(output_dir, 'interaction_plot.png'))
+
+    # Process statements
+    # This transforms the statement information into a format that can be used to
+    # render statements using the Vue.js components on the frontend
+    stmt_data_per_gene = {}
+    for name, gene_stmts_df in filtered_df.groupby('name'):
+        stmt_jsons = []
+        for sj in gene_stmts_df["stmt_json"].values:
+            if isinstance(sj, str):
+                try:
+                    stmt_jsons.append(json.loads(sj))
+                except json.JSONDecodeError as e:
+                    print(f"Failed to decode JSON string: {sj[:100]}... -> {e}")
+                    continue
+            elif isinstance(sj, dict):
+                stmt_jsons.append(sj)
+            else:
+                print(f"Unexpected type in stmt_json column: {type(sj)} -> {sj}")
```

Comment:
Maybe transform the `print` statements to `logger.warning` and/or `logger.error`
```suggestion
                    logger.warning(f"Failed to decode JSON string: {sj[:100]}... -> {e}")
                    continue
            elif isinstance(sj, dict):
                stmt_jsons.append(sj)
            else:
                logger.warning(f"Unexpected type in stmt_json column: {type(sj)} -> {sj}")
```